### PR TITLE
[Core] Clean up `KratosComponents`

### PR DESCRIPTION
**Pull request by loumalouomega**
**📝 Description**

This PR refactors the `KratosComponents` class in the `kratos_components.h` file. The changes include reordering and reformatting the code, adding documentation comments. The class now has improved readability and maintainability.

**🆕 Changelog**

- [Clean up `KratosComponents`](https://github.com/KratosMultiphysics/Kratos/commit/76e88eed98c9a7d22195b5bc7ede2bf806dbfc1e)

### CODE REVIEW

**Review by RiccardoRossi — COMMENTED** [review]
this is VERY delicate due to linking in the different applications. 
honestly i would not touch this, particularly thinking that the new registry mechanism is currently being ramped up. Having said this, i leave the word to @roigcarlo

File: kratos/includes/kratos_components.h [hunk 1]
```diff
@@ -232,21 +228,19 @@ class KRATOS_API(KRATOS_CORE) KratosComponents
...
     ///@{
 
-
     ///@}
     ///@name Un accessible methods
     ///@{
 
     /// Assignment operator.
-    KratosComponents& operator=(KratosComponents const& rOther);
+    KratosComponents& operator=(KratosComponents& rOther);
```
Comment (roigcarlo) [inline kratos/includes/kratos_components.h]: Missing const.

File: kratos/includes/kratos_components.h [hunk 2]
```diff
@@ -232,21 +228,19 @@ class KRATOS_API(KRATOS_CORE) KratosComponents
...
 
     /// Copy constructor.
-    KratosComponents(KratosComponents const& rOther);
+    KratosComponents(KratosComponents& rOther);
```
Comment (roigcarlo) [inline kratos/includes/kratos_components.h]: Missing const.

File: kratos/includes/kratos_components.h [hunk 3]
```diff
@@ -453,21 +449,19 @@ class KRATOS_API(KRATOS_CORE) KratosComponents<VariableData>
...
     ///@{
 
-
     ///@}
     ///@name Un accessible methods
     ///@{
 
     /// Assignment operator.
-    KratosComponents& operator=(KratosComponents const& rOther);
+    KratosComponents& operator=(KratosComponents& rOther);
```
Comment (roigcarlo) [inline kratos/includes/kratos_components.h]: Missing const

File: kratos/includes/kratos_components.h [hunk 4]
```diff
@@ -453,21 +449,19 @@ class KRATOS_API(KRATOS_CORE) KratosComponents<VariableData>
...
 
     /// Copy constructor.
-    KratosComponents(KratosComponents const& rOther);
+    KratosComponents(KratosComponents& rOther);
```
Comment (roigcarlo) [inline kratos/includes/kratos_components.h]: Missing const

File: kratos/includes/kratos_components.h [hunk 5]
```diff
@@ -508,25 +502,23 @@ inline std::ostream& operator << (std::ostream& rOStream,
...
+void KRATOS_API(KRATOS_CORE) AddKratosComponent(const std::string& rName, const Variable<unsigned int>& rComponent);
+void KRATOS_API(KRATOS_CORE) AddKratosComponent(const std::string& rName, const Variable<double>& rComponent);
+void KRATOS_API(KRATOS_CORE) AddKratosComponent(const std::string& rName, const Variable<array_1d<double, 3>>& rComponent);
+void KRATOS_API(KRATOS_CORE) AddKratosComponent(const std::string& rName, const Variable<array_1d<double, 4>>& rComponent);
+void KRATOS_API(KRATOS_CORE) AddKratosComponent(const std::string& rName, const Variable<array_1d<double, 6>>& rComponent);
+void KRATOS_API(KRATOS_CORE) AddKratosComponent(const std::string& rName, const Variable<array_1d<double, 9>>& rComponent);
+void KRATOS_API(KRATOS_CORE) AddKratosComponent(const std::string& rName, const Variable<Quaternion<double>>& rComponent);
+void KRATOS_API(KRATOS_CORE) AddKratosComponent(const std::string& rName, const Variable<Vector>& rComponent);
+void KRATOS_API(KRATOS_CORE) AddKratosComponent(const std::string& rName, const Variable<Matrix>& rComponent);
+void KRATOS_API(KRATOS_CORE) AddKratosComponent(const std::string& rName, const Variable<std::string>& rComponent);
```
Comment (roigcarlo) [inline kratos/includes/kratos_components.h:516]: Missing const on the rComponent of al that entries.

Comment (roigcarlo) [reply kratos/includes/kratos_components.h:516]: A no, those are there mb.

File: kratos/includes/kratos_components.h (hunk 2 above)
Comment (loumalouomega) [reply kratos/includes/kratos_components.h]: Done

File: kratos/includes/kratos_components.h (hunk 1 above)
Comment (loumalouomega) [reply kratos/includes/kratos_components.h]: Done

File: kratos/includes/kratos_components.h (hunk 4 above)
Comment (loumalouomega) [reply kratos/includes/kratos_components.h]: Done

File: kratos/includes/kratos_components.h (hunk 3 above)
Comment (loumalouomega) [reply kratos/includes/kratos_components.h]: Done

**Review by roigcarlo — APPROVED** [review]
Ok, since there are only cosmetics, but please do not touch this kind of classes in July when there is hardly anyone to review in detail.

**Comment by loumalouomega** [comment]
> Ok, since there are only cosmetics, but please do not touch this kind of classes in July when there is hardly anyone to review in detail.

image

**Pull request merged**